src/librustc_mir/interpret/intrinsics/caller_location.rs

```diff
@@ -6,6 +6,28 @@ use syntax_pos::{Symbol, Span};
 use crate::interpret::{Scalar, MemoryKind, MPlaceTy, intrinsics::{InterpCx, Machine}};
 
 impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
+    /// Walks up the callstack from the intrinsic's callsite, searching for the first frame which is
+    /// not `#[track_caller]`. Returns the (passed) span of the intrinsic's callsite if the first
+    /// frame in the stack is untracked so that we can display the callsite of the intrinsic within
+    /// that function.
+    crate fn find_closest_untracked_caller_location(
+        &self,
+        intrinsic_loc: Span,
+    ) -> Span {
```

Comment:
You could make this return `Option<Span>` to avoid needing to take in the `intrinsic_loc`.